src/bin/smoosh_tools.rs

```diff
@@ -0,0 +1,787 @@
+
+    Ok(())
+}
+
+fn update_cargo<T>(cargo: &PathBuf, replacer: T) -> Result<(), Error>
+where
+    T: DependencyLineReplacer,
+{
+    let content = read_file(cargo)?;
+    let mut filtered_lines = Vec::new();
+    for line in content.split("\n") {
+        if line.starts_with(
+            "# jsparagus = { git = \"https://github.com/mozilla-spidermonkey/jsparagus\",",
+        ) || line.starts_with(
+            "jsparagus = { git = \"https://github.com/mozilla-spidermonkey/jsparagus\",",
+        ) {
+            let orig_line = if line.starts_with("# ") {
+                &line[2..]
+            } else {
+                line
+            };
+            replacer.on_official(orig_line, &mut filtered_lines)
+        } else if line.starts_with("jsparagus = ") {
+        } else {
+            filtered_lines.push(line.to_string());
+        }
+    }
+    write_file(cargo, filtered_lines.join("\n"))
+}
+
+fn run_mach(command: &'static str, args: &SimpleArgs) -> Result<(), Error> {
+    let mc = MozillaCentralTree::try_new(&args.mc_path)?;
+    let jsparagus = JsparagusTree::try_new()?;
+    let mozconfig = jsparagus.mozconfig(args.build_type);
+
+    let args = &[command];
+    let cmdline = format_cmdline("mach", args);
+    log_info_cmd!(cmdline);
+    check_command(
+        Command::new(mc.topsrcdir.join("mach").to_str().unwrap())
+            .args(args)
+            .current_dir(mc.topsrcdir)
+            .env("MOZCONFIG", mozconfig.to_str().unwrap()),
+        cmdline,
+    )
+}
+
+fn build(args: &SimpleArgs) -> Result<(), Error> {
+    let mc = MozillaCentralTree::try_new(&args.mc_path)?;
+    let jsparagus = JsparagusTree::try_new()?;
+
+    update_cargo(
+        &mc.smoosh_cargo,
+        LocalDependencyLineReplacer {
+            jsparagus: jsparagus.topsrcdir,
+        },
+    )?;
+
+    run_mach("build", args)
+}
+
+fn shell(args: &SimpleArgs) -> Result<(), Error> {
+    run_mach("run", args)
+}
+
+fn test(args: &SimpleArgs) -> Result<(), Error> {
+    run_mach("jstests", args)?;
+    run_mach("jit-test", args)
+}
+
+fn vendor(mc: &MozillaCentralTree) -> Result<(), Error> {
+    let args = &["vendor", "rust"];
+    let cmdline = format_cmdline("mach", args);
+    log_info_cmd!(cmdline);
+    check_command(
+        Command::new(mc.topsrcdir.join("mach").to_str().unwrap())
+            .args(args)
+            .current_dir(mc.topsrcdir.clone()),
+        cmdline,
+    )
+}
+
+fn bump(args: &SimpleArgs) -> Result<(), Error> {
+    let mc = MozillaCentralTree::try_new(&args.mc_path)?;
+    let jsparagus = JsparagusTree::try_new()?;
+
+    let jsparagus_repo = GitRepository::try_new(jsparagus.topsrcdir.clone())?;
+
+    log_info!("Checking ci_generated branch HEAD");
+
+    jsparagus_repo.ensure_remote(
+        "upstream",
+        "https://github.com/mozilla-spidermonkey/jsparagus.git",
+    )?;
+    let remotes = jsparagus_repo.ls_remote("upstream")?;
```

Comment:
nit: `jsparagus_repo.ls_remote("https://github.com/mozilla-spidermonkey/jsparagus.git")`

This way this would not change the list of remotes of the users.